PubMed.js

```diff
@@ -139,45 +140,43 @@ function scrapeItemProps(itemprops) {
  * My Bibliography
  */
 function getSearchResults(doc, checkOnly) {
-	var results = doc.getElementsByClassName('rslt');
-	if (!results.length) {
-		//My Bibliography
-		results = doc.getElementsByClassName('citationListItem');
-	}
-	
-	if (!results.length) return false;
-	
+	var results = doc.querySelectorAll('.rslt, .docsum-wrap, citationListItem');
 	var items = {}, found = false;
 	
-	for (var i=0; i<results.length; i++) {
-		var title = ZU.xpathText(results[i], '(.//p[@class="title"]|.//h1)[1]')
-			|| ZU.xpathText(results[i], './div[@class="docsumRightcol"]/a'); //My Bibliography
-		
-		var uid = ZU.xpathText(results[i], './/input[starts-with(@id,"UidCheckBox")]/@value')
-			|| ZU.xpathText(results[i], './div[@class="chkBoxLeftCol"]/input/@refuid') //My Bibliography
-			|| ZU.xpathText(results[i], './/dl[@class="rprtid"]/dd[preceding-sibling::*[1][text()="PMID:"]]');
-		
-		if (!uid) {
-			uid = ZU.xpathText(results[i], './/p[@class="title"]/a/@href');
-			if (uid) uid = uid.match(/\/(\d+)/);
-			if (uid) uid = uid[1];
-		}
-		
-		if (!uid || !title) continue;
-		
-		if (checkOnly) return true;
-		found = true;
-		
-		// Checkbox is a descendant of the containing .rprt div
-		var checkbox = ZU.xpath(results[i].parentNode, './/input[@type="checkbox"]')[0];
+	if (!results.length) return false;
 
-		// Keys must be strings. Otherwise, Chrome sorts numerically instead of by insertion order.
-		items["u"+uid] = {
-			title: ZU.trimInternal(title),
-			checked: checkbox && checkbox.checked
-		};	
+	if (results.length > 0){
```

Comment:
Isn't it enough to either do the test on line 146 and fail when length is zero or condition what is following to the fact that length is positive?